### PR TITLE
Revert card style design for messages to a 1px divider

### PR DESCRIPTION
**Pull request by LunarX**
Depends on https://github.com/Infomaniak/android-core/pull/33

### CODE REVIEW

**Review by JorisBodin — APPROVED** [review]

File: app/src/main/res/layout/item_message.xml [hunk 1]
```diff
@@ -15,468 +15,434 @@
   ~ You should have received a copy of the GNU General Public License
   ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   -->
-<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
+
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/item_message.xml:18]:
```suggestion
```

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 2]
```diff
@@ -109,16 +109,18 @@ object UiUtils {
         @ColorInt newColor: Int,
         duration: Long = 150L,
         animate: Boolean = true,
-        setColor: (Int) -> Unit,
-    ) {
+        setColor: (color: Int) -> Unit,
+    ): ValueAnimator? {
         if (animate) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:114]:
```suggestion
        return if (animate) {
```

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 3]
```diff
@@ -109,16 +109,18 @@ object UiUtils {
...
-            ValueAnimator.ofObject(ArgbEvaluator(), oldColor, newColor).apply {
+            return ValueAnimator.ofObject(ArgbEvaluator(), oldColor, newColor).apply {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:115]:
```suggestion
            ValueAnimator.ofObject(ArgbEvaluator(), oldColor, newColor).apply {
```

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 4]
```diff
@@ -109,16 +109,18 @@ object UiUtils {
...
                 setDuration(duration)
                 addUpdateListener { animator -> setColor(animator.animatedValue as Int) }
                 start()
             }
         } else {
             setColor(newColor)
         }
+
+        return null
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:124]:
```suggestion
            null
        }
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 5]
```diff
@@ -245,9 +285,15 @@ class ThreadFragment : Fragment() {
         // TODO: When opening a Thread via a Notification, the action of leaving this fragment
         // TODO: (either via a classic Back button, or via this `popBackStack`) will probably
         // TODO: do nothing instead of going back to the ThreadList fragment (as it should be).
+        valueAnimator?.cancel()
         findNavController().popBackStack()
     }
 
+    enum class HeaderState {
+        ELEVATED,
+        LOWERED
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:294]:
```suggestion
        LOWERED,
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 6]
```diff
@@ -82,23 +95,47 @@ class ThreadFragment : Fragment() {
...
-        toolbar.setNavigationOnClickListener { findNavController().popBackStack() }
+        toolbar.setNavigationOnClickListener { leaveThread() }
 
         val defaultTextColor = context.getColor(R.color.primaryTextColor)
         appBar.addOnOffsetChangedListener { appBarLayout, verticalOffset ->
             val total = appBarLayout.height * COLLAPSE_TITLE_THRESHOLD
             val removed = appBarLayout.height - total
-            val progress = ((-verticalOffset.toFloat()) - removed).coerceAtLeast(0.0)
-            val opacity = ((progress / total) * 255).roundToInt()
+            val progress = (((-verticalOffset.toFloat()) - removed).coerceAtLeast(0.0) / total).toFloat() // [0,1]
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:109]: Replace with a real explaining comment.

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**